src/wp-includes/class-wp-html-attribute-token.php

```diff
@@ -0,0 +1,89 @@
+<?php
```

Comment:
@dmsnell @azaozz What do you think of moving the API's files into `wp-includes/html-api/` folder, similar to other APIs such as REST?

Reply:
I considered this, though I considered the `html` directory name instead of `html-api`, but decided to leave it flat at the top given how most existing libraries, even ones with multiple modules, are there vs. in their own directories.

looking towards the future I think it's reasonable to expect between one and three more classes to support the HTML system, but unlikely more than that.

Reply:
Moving to a directory makes sense imho. Many new APIs land in separate sub-directories of wp-includes. Not so sure about the name though. Both `html` and html-api` sound pretty commonplace/non-specific? Sounds like most functionality that's in formatting.php (`wpautop()`, `texturize()`, etc.) should be in there, right?

On the other hand, can't think of a better name either. `html-tag-processor` might be okay? But `html-api` is probably better?

Reply:
For example, the Fonts API will be in its own directory to group the API's files into one "package". Might be worth considering doing the same for the HTML API. In that way, it's clearly communicating to contributors and extenders that this API is composed of these files. Think of the directory as a grouping mechanism.

Reply:
> html-tag-processor might be okay? 

@azaozz there will at least be `wp-html-processor.php` coming, so I don't want to scope this too narrowly to the tag processor.

> Sounds like most functionality that's in formatting.php (wpautop(), texturize(), etc.) should be in there, right?

this is something I'm not sure I have a feeling for yet. those functions serve different purposes and they aren't particularly tied to HTML, rather they are tied to WordPress and how it processes input.

the primary goal I want to communicate here is that these modules deal specifically with HTML and the modes of interacting with it. if this goes well and continues to see adoption, in the eventuality that functions like `wpautop()` and `texturize()` utilize this code, I would predict they would stay right where they are and make calls to the tag processor or html processor as part of how they work.

---

> Might be worth considering…

@hellofromtonya are you saying you prefer this go in a new directory or suggesting we consider? as stated above, I shared my rationale for why, when we considered this before, we chose what we did.

to elaborate, my slight preference for `html` over `html-api` is that the `-api` part is redundant, and most existing libraries/packages don't include it, namely everything except the `rest-api` library, which itself is following more common nomenclature whereas people tend to call it a "REST API" vs. "the REST library" or "the REST code."

not a thing I care much about, just repeating the rationale in case I was unclear earlier.

Reply:
@hellofromtonya I've moved these into a new `html` directory. if you have qualms with that name we can change it.

Reply:
Thanks for moving them @dmsnell. I agree with @azaozz that the name `html` has broad meaning in Core and beyond. It's too generic, which could be confusing of what should be in that directory.

I'd vote for renaming it to `html-api` so that it clearly documents that this directory contains all of the HTML API files and only the API's files.

I know it seems like a nit-pick. Adding the `-api` helps to communicate and document what should and shouldn't be inside. That can help share context for contributors and extenders today and into the future.

Reply:
while I don't follow the logic of the argument, I have changed the name to `HTML-API` in 361710d91056320ca2fd2e232e02b8fa6fdde17a